Source/JavaScriptCore/jsc.cpp

```diff
@@ -1389,12 +1389,25 @@ JSC_DEFINE_HOST_FUNCTION(functionAtob, (JSGlobalObject* globalObject, CallFrame*
 {
     VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
-    auto* string = callFrame->argument(0).toString(globalObject);
+
+    if (!callFrame->argumentCount())
+        return JSValue::encode(throwException(globalObject, scope, createError(globalObject, "Missing input for atob."_s)));
+
+    JSValue jsValue = callFrame->argument(0);
+    if (jsValue.isUndefined())
+        return JSValue::encode(throwException(globalObject, scope, createError(globalObject, "InvalidCharacterError in argument for atob."_s)));
```

Comment:
If you want two distinct errors, you can keep things as is. But "argument(...)" does a bounds check. So you could remove the above "!argumentCount" check and just keep this, and rename the error to something like "atob must take XYZ arguments" whatever XYZ is. And that way you'd end up with less code in these functions in the shell

Reply:
Yijia, I didn't mean that you literally need to have `InvalidCharacterError` in the error string.  Your original error message is fine.  Or you can do what Saam said and combine these checks and use a single error message.  If you do that, just `"InvalidCharacterError"_s` is fine too.